ape_solidity/compiler.py

```diff
@@ -368,78 +368,41 @@ def get_version_map(
         contracts_path = base_path or self.config_manager.contracts_folder
         imports = self.get_imports(get_all_files_in_directory(contracts_path), contracts_path)
 
-        def get_imported_source_paths(
-            path: Path, source_ids_checked: Optional[List[str]] = None
-        ) -> Set[Path]:
-            source_ids_checked = source_ids_checked or []
-            source_id = str(get_relative_path(path, contracts_path))
-            if source_id in source_ids_checked:
-                # Already got this source's imports
-                return set()
-
-            source_ids_checked.append(source_id)
-            import_paths = [contracts_path / i for i in imports.get(source_id, []) if i]
-            return_set = {i for i in import_paths}
-            for import_path in import_paths:
-                indirect_imports = get_imported_source_paths(
-                    import_path, source_ids_checked=source_ids_checked
-                )
-                for indirect_import in indirect_imports:
-                    return_set.add(indirect_import)
-
-            return return_set
-
         # Add imported source files to list of contracts to compile.
         source_paths_to_compile = {p for p in contract_filepaths}
         for source_path in contract_filepaths:
-            imported_source_paths = get_imported_source_paths(source_path)
+            imported_source_paths = self._get_imported_source_paths(
+                source_path, contracts_path, imports
+            )
             for imported_source in imported_source_paths:
                 source_paths_to_compile.add(imported_source)
 
         # Use specified version if given one
         if self.config.version is not None:
-            return {Version(self.config.version): source_paths_to_compile}
-        # else: find best version per source file
+            specified_version = Version(self.config.version)
+            if specified_version not in self.installed_versions:
```

Comment:
@helloibis this was the problem here- when specifying a version, and that version was not installed, it would fail. So it was specific to that scenario.

I also refactors to try and make things easier and corrected some Pycharmisms